static/app/views/integrationPipeline/githubInstallationSelect.tsx

```diff
@@ -0,0 +1,134 @@
+import {Fragment, useState} from 'react';
+import styled from '@emotion/styled';
+import * as qs from 'query-string';
+
+import {addLoadingMessage} from 'sentry/actionCreators/indicator';
+import {BaseAvatar} from 'sentry/components/core/avatar/baseAvatar';
+import {Button} from 'sentry/components/core/button';
+import type {SelectKey, SelectOption} from 'sentry/components/core/compactSelect';
+import {CompactSelect} from 'sentry/components/core/compactSelect';
+import Form from 'sentry/components/forms/form';
+import FormModel from 'sentry/components/forms/model';
+import List from 'sentry/components/list';
+import ListItem from 'sentry/components/list/listItem';
+import {t} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+
+type Installation = {
+  avatar_url: string;
+  github_organization: string;
+  installation_id: number;
+};
+
+type GithubInstallationProps = {
+  installation_info: Installation[];
+};
+
+export function GithubInstallationSelect({installation_info}: GithubInstallationProps) {
+  const [model] = useState<FormModel>(() => new FormModel());
+  const [installationID, setInstallationID] = useState<SelectKey | null>(null);
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+
+  const handleSubmit = (e: React.MouseEvent) => {
+    e.preventDefault();
+    const data = model.getData();
+    addLoadingMessage(t('Submitting'));
+    setIsSaving(true);
+    model.setFormSaving();
+
+    const {
+      location: {origin},
+    } = window;
+
+    // redirect to the extensions endpoint with the form fields as query params
+    // this is needed so we don't restart the pipeline loading from the original
+    // OrganizationIntegrationSetupView route
+    const chosen_installation_id = data.installation_id || -1;
+    const currentParams = new URLSearchParams(window.location.search);
+    const newParams = {
+      ...Object.fromEntries(currentParams),
+      chosen_installation_id,
+    };
+
+    const newUrl = `${origin}/extensions/github/setup/?${qs.stringify(newParams)}`;
+    return window.location.assign(newUrl);
```

Comment:
We can simplify this all with the `useNavigate` hook

```tsx
const location = useLocation()
const navigate = useNavigate()
navigate({pathnam: `/extensions/github/setup`, query: {...location.query, chosen_installation_id: data.installation_id || 1})
```
or something similar. Try not to use `window` if we can avoid it, but I'm not sure how these views in pipelines behave.

Reply:
so I struggled through this for an afternoon but I think we have to use `window.location.assign ` (?). Because we need a hard reload/redirect, unless there's alternative way (pls inform) :hides:

Some notes:

- Setting up the `useNavigate` & `useLocation` to work with the flow means we'll need to change the `useMemoryRouter` -> `BrowserRouter` , because apparently `MemoryRouter` doesn't store location information for `useLocation` ??? 
- The main issue I ran into was `navigate` would put the correct redirect url into the browser but would not actually load/redirect to the URL. I'm not too familiar with routers, but I hypothesize that a router cannot route to somewhere that's not a part of the react app (??).

Reply:
Makes sense, thanks for looking into it! `window.assign` isn't actually bad to use I just wanted to double check if we had to or not, but it seems like we do 👍  thanks for taking the time to dig!